internal/lint/error.go

```diff
@@ -0,0 +1,558 @@
+// or if there is no var statements at all - the end of the "import" statement
+func (c currentInfo) varEnd() token.Pos {
+	if c.varNode == nil {
+		return c.importEnd()
+	}
+
+	return c.varNode.End()
+}
+
+// currentInfos holds current information for each golang file
+type currentInfos struct {
+	info map[string]*currentInfo
+}
+
+// entry creates or gets the current info for a file path
+func (c *currentInfos) entry(path string) *currentInfo {
+	if c.info == nil {
+		c.info = map[string]*currentInfo{}
+	}
+
+	var i *currentInfo
+	var ok bool
+	if i, ok = c.info[path]; !ok {
+		i = &currentInfo{}
+		c.info[path] = i
+	}
+
+	return i
+}
+
+// appendErrors adds given errors to the list of errors for the file path
+func (c *currentInfos) appendErrors(path string, errors ...string) {
+	c.entry(path).appendErrors(errors...)
+}
+
+// setPackage sets the position of the "package" statement token
+func (c *currentInfos) setPackage(path string, pos token.Pos) {
+	c.entry(path).packagePos = pos
+}
+
+// setImport sets the position of the "import" statement token and checks to see
+// if the error package is already imported making note of it's imported name
+func (c *currentInfos) setImport(path string, node *ast.GenDecl) {
+	entry := c.entry(path)
+	for _, i := range node.Specs {
+		imp := i.(*ast.ImportSpec)
+		if imp.Path.Value == `"`+errorPackage+`"` {
+			// there can be only one import to the error package
+			entry.name = imp.Name.Name
+		}
+	}
+
+	// if there are more than one import statements last one wins
+	entry.importNode = node
+}
+
+// setPackage sets the position of the "var" statement token
+func (c *currentInfos) setVar(path string, node *ast.GenDecl) {
+	c.entry(path).varNode = node
+}
+
+// allErrors returns all errors for a given package
+func (c *currentInfos) allErrors() []string {
+	allErrors := make([]string, 0, 10)
+	for _, v := range c.info {
+		allErrors = append(allErrors, v.errors...)
+	}
+
+	return allErrors
+}
+
+// current Analyzer scans the package files to gather the current error
+// information. We need this first pass to gather positional data and a list of
+// defined errors within the package
+var current = analysis.Analyzer{
+	Name:       "error_current",
+	Doc:        "Fetches the currently defined errors",
+	Run:        currentErrors,
+	Requires:   []*analysis.Analyzer{inspect.Analyzer},
+	ResultType: reflect.TypeOf((*currentInfos)(nil)),
+}
+
+// errorVariableDecl determines what errors are defined in a "var" statement and
+// returns if any are
+func errorVariableDecl(pass *analysis.Pass, d ast.Decl) ([]string, bool) {
+	var g *ast.GenDecl
+	var ok bool
+
+	if g, ok = d.(*ast.GenDecl); !ok || g.Tok != token.VAR {
+		return nil, false
+	}
+
+	names := make([]string, 0, 5)
+	for _, s := range g.Specs {
+		var spec *ast.ValueSpec
+
+		if spec, ok = s.(*ast.ValueSpec); !ok {
+			continue
+		}
+
+		if len(spec.Values) != 1 {
+			continue
+		}
+
+		val := spec.Values[0]
```

Comment:
nit: I think not having an empty line before this statement would improve the readability of the code, as it would group it with the relevant if-block.